x/profiles/keeper/grpc_query.go

```diff
@@ -122,6 +123,43 @@ func (k Keeper) ChainLinks(ctx context.Context, request *types.QueryChainLinksRe
 	return &types.QueryChainLinksResponse{Links: links, Pagination: pageRes}, nil
 }
 
+// ChainLinkOwners implements the Query/ChainLinkOwners gRPC method
+func (k Keeper) ChainLinkOwners(ctx context.Context, request *types.QueryChainLinkOwnersRequest) (*types.QueryChainLinkOwnersResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	store := sdkCtx.KVStore(k.storeKey)
+
+	ownersPrefix := types.ChainLinkChainPrefix
+	switch {
+	case request.ChainName != "" && request.Target != "":
+		ownersPrefix = types.ChainLinkChainAddressKey(request.ChainName, request.Target)
+	case request.ChainName != "":
+		ownersPrefix = types.ChainLinkChainKey(request.ChainName)
+	}
+
+	var owners []types.QueryChainLinkOwnersResponse_ChainLinkOwnerDetails
+	ownersStore := prefix.NewStore(store, ownersPrefix)
+	pageRes, err := query.Paginate(ownersStore, request.Pagination, func(key []byte, value []byte) error {
+		keyWithPrefix := append(ownersPrefix, key...)
+		cleanedKey := bytes.TrimSuffix(bytes.TrimPrefix(keyWithPrefix, types.ChainLinkChainPrefix), value)
+		values := bytes.Split(cleanedKey, types.Separator)
+		chainName, target := values[0], values[1]
```

Comment:
I think it can be all implemented by `bytes.Split` instead of `bytes.TrimSuffix`, say:
```go
values := bytes.Split(cleanedKey, types.Separator)
chainName, target, value := values[0], values[1], values[2]
```
Then, the value of the key can be simply `0x1` in order to reduce the storage on chain.

Reply:
Right now, links owners are stored using the following keys: 
```
ChainLinkChainPrefix + Chain name + Separator + External Address + User address -> User address
```

So, if we use the `bytes.Split` it would return the following values:
```
1. Chain name
2. External Address + User address
```

We would have to remove the `User address` suffix anyway to get the external address

Reply:
@RiccardoM I mean that we can store the key like:

```
ChainLinkChainPrefix + Chain name + Separator + External Address + Spearator + User address -> 0x01
```

So the storage doesn't need to store user address again.

Reply:
@dadamu You are completely right! Thanks for pointing that out, this way we can save up a lot of disk space. I've just implemented it this way (and also added some missing tests)

Reply:
Considering this answer, should we look after other places where we use `bytes.TrimSuffix` and `bytes.TrimPrefix` and see if we can improve them as well?

Reply:
@bragaz All other keys instances do not trim prefixes unless necessary (eg. inside user group members). All the keys have already been reviewed inside their own PRs before being merged, and I don't think it would be appropriate to do it here anyway. I've checked though and didn't find any use of `bytes.TrimPrefix` aside from this case and user groups members anyway